includes/payment-methods/class-wc-stripe-upe-payment-method-bacs-debit.php

```diff
@@ -93,6 +98,39 @@ public function create_payment_token_for_user( $user_id, $payment_method ) {
 		return $token;
 	}
 
+	/**
+	 * Hides the Bacs payment method for pre-orders that are charged upon release.
+	 *
+	 * This function removes the "stripe_bacs_debit" payment gateway from the available payment methods
+	 * if the cart contains a pre-order product that is charged upon release. This ensures that customers
+	 * do not select Bacs for pre-orders that are not charged upfront.
+	 *
+	 * @return void
+	 */
+	public function hide_bacs_for_pre_orders_charge_upon_release() {
+		add_filter(
+			'woocommerce_available_payment_gateways',
+			function ( $available_gateways ) {
+				if ( is_checkout() && WC_Pre_Orders_Cart::cart_contains_pre_order() ) {
+					// Iteration is unnecessary since only one pre-order product can be in the cart.
+					$product_id = reset( WC()->cart->get_cart() )['product_id'];
+					if ( WC_Pre_Orders_Product::product_is_charged_upon_release( $product_id ) ) {
+						unset( $available_gateways['stripe_bacs_debit'] );
+					}
+				}
+				return $available_gateways;
+			}
+		);
+	}
+
+	/**
+	 * Hides the Bacs payment method for subscriptions with free trials.
+	 *
+	 * This function removes the "stripe_bacs_debit" payment gateway from the available payment methods
+	 * if the cart contains a subscription with a free trial and the total cart amount is zero.
+	 *
+	 * @return void
+	 */
```

Comment:
This should have been included in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3987.